src/app/components-small/plugin-filter-node/plugin-filter-node.component.ts

```diff
@@ -36,47 +39,47 @@ export class PluginFilterNodeComponent implements OnInit {
             console.warn("No filter object provided to plugin filter node component");
             return;
         }
-        const isLeaf = (filter.name != null) || filter.tag || filter.version;
+        // Should filters have multiple attributes at some point, this must be changed
+        const type = Object.keys(filter)[0];
+        if (type !== 'and' && type !== 'or' && type !== 'name' && type !== 'tag' && type !== 'version') {
+            console.warn("Invalid filter type provided to plugin filter node component");
+            return;
+        }
+        this.type = type as FilterType;
+        const isLeaf = filter.and == null && filter.or == null;
         if (isLeaf) {
-            this.type = filter.version ? 'version' : filter.tag ? 'tag' : 'name';
-            this.value = filter.name ?? filter.tag ?? filter.version;
+            this.value = filter[this.type];
         } else {
-            this.type = filter.and ? 'and' : 'or';
-            this.children = filter.and ?? filter.or;
+            this.children = filter[this.type];
         }
         this.filterObject = currentValue;
         this.childChange.emit([this.index, this.filterObject]);
     }
 
-    setFilter(type: 'name' | 'tag' | 'version' | 'and' | 'or') {
-        const isLeaf = type === 'name' || type === 'tag' || type === 'version';
+    setFilter(type: FilterType) {
+        const isLeaf = type !== 'and' && type !== 'or';
```

Comment:
It is probably a good idea to add a comment to remind future developers why this (and other locations) do not test for the "not" filter. The best location for this is probably the definition of the type "FilterType"